### PR TITLE
feat(sma): implement v2 POST /operation API

### LINKED ISSUE #3496: [v2 sys-mgmt-agent] Implement v2 POST /operation API
# 🚀 Feature Request

### Relevant Package

 This feature request is for sys-mgmt-agent

### Description
Implement v2 POST `/operation` API

### PR DESCRIPTION
**Pull request by hahattan**
Signed-off-by: Chris Hung <chris@iotechsys.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## What is the new behavior?
example request body POSTing `/api/v2/operation`:
```
[
    {
        "apiVersion": "v2",
        "requestId": "2f6fe5e2-dd93-478c-a2ba-21642e42a5f5",
        "serviceName": "edgex-core-data",
        "action": "start"
    },
    {
        "apiVersion": "v2",
        "requestId": "998d70a3-7245-4735-b308-ebb9f448fb38",
        "serviceName": "edgex-core-command",
        "action": "start"
    }
]
```

example response:
```
207 Multi-Status
[
    {
        "apiVersion": "v2",
        "requestId": "998d70a3-7245-4735-b308-ebb9f448fb38",
        "statusCode": 200
    },
    {
        "apiVersion": "v2",
        "requestId": "2f6fe5e2-dd93-478c-a2ba-21642e42a5f5",
        "statusCode": 200
    }
]
```

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information
@lenny-intel are we planning to remove the `edgex-` prefix for container name as well ?
The behavior in SMA is inconsistent:
 - for API that relies on executor (e.g. `/operation`, `metrics/{services}`), you need to refer the service by its container name, which have `edgex-` prefix
 - for API which will trigger requested service's own API (e.g. `/configs/{services}`), you don't have to add `edgex-` prefix

For example:
```
$ curl http://localhost:58890/api/v2/configs/core-data,core-metadata

200
{...}

$ curl  http://localhost:58890/api/v2/metrics/core-data,core-metadata
200 (should be 400 or 404 but need to refactor sys-mgmt-executor ....)
{
    "apiVersion": "v2",
    "statusCode": 200,
    "metrics": {
        "core-data": {
            "Success": false,
            "errorMessage": "exit status 1",
            "executor": "docker",
            "operation": "metrics",
            "service": "core-data"
        },
        "core-metadata": {
            "Success": false,
            "errorMessage": "exit status 1",
            "executor": "docker",
            "operation": "metrics",
            "service": "core-metadata"
        }
    }
}

curl http://localhost:58890/api/v2/metrics/edgex-core-data,edgex-core-metadata
200
{
    "apiVersion": "v2",
    "statusCode": 200,
    "metrics": {
        "edgex-core-data": {
            "Success": true,
            "executor": "docker",
            "operation": "metrics",
            "result": {
                "cpuUsedPercent": 0.02,
                "memoryUsed": 7184843,
                "raw": {
                    "block_io": "0B / 0B",
                    "cpu_perc": "0.02%",
                    "mem_perc": "0.34%",
                    "mem_usage": "6.852MiB / 1.94GiB",
                    "net_io": "46.9kB / 46kB",
                    "pids": "12"
                }
            },
            "service": "edgex-core-data"
        },
        "edgex-core-metadata": {
            "Success": true,
            "executor": "docker",
            "operation": "metrics",
            "result": {
                "cpuUsedPercent": 0.06,
                "memoryUsed": 7679771,
                "raw": {
                    "block_io": "0B / 0B",
                    "cpu_perc": "0.06%",
                    "mem_perc": "0.37%",
                    "mem_usage": "7.324MiB / 1.94GiB",
                    "net_io": "42.1kB / 40.8kB",
                    "pids": "12"
                }
            },
            "service": "edgex-core-metadata"
        }
    }
}
```

### CODE REVIEW

**Comment by hahattan** [comment]
recheck

**Comment by lenny-intel** [comment]
recheck

**Review by cloudxxx8 — DISMISSED** [review]
LGTM

File: internal/system/agent/v2/application/executor/operation.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+// NewOperation is a factory function that returns an initialized operation receiver struct.
+func NewOperation(executor interfaces.CommandExecutor, executorPath string, lc logger.LoggingClient) *operation {
+	return &operation{
+		executor:     executor,
+		executorPath: executorPath,
+		lc:           lc,
+	}
+}
+
+// Do implements the Metrics interface to obtain metrics via executor for one or more services concurrently.
```
Comment (lenny-intel) [inline internal/system/agent/v2/application/executor/operation.go:37]: I don't think this comment correct, i.e. specific to metrics? Please re-word the comment.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by lenny-intel — APPROVED** [review]
LGTM

**Comment by hahattan** [comment]
@lenny-intel btw do you have any comment about what I said in 'Other Information' in the PR  ?

**Comment by lenny-intel** [comment]
> are we planning to remove the `edgex-` prefix for container name as well ?
> The behavior in SMA is inconsistent:
> 
> * for API that relies on executor (e.g. `/operation`, `metrics/{services}`), you need to refer the service by its container name, which have `edgex-` prefix
> * for API which will trigger requested service's own API (e.g. `/configs/{services}`), you don't have to add `edgex-` prefix

We did not plan to make that change to the container names, but I do see the inconsistency that it creates.
@jpwhitemn , please add this to next weeks Core WG agenda, i.e. **Remove `edgex-` prefix from container names in compose files**

**Comment by hahattan** [comment]
@lenny-intel do we have any decision about the container naming ?

**Comment by lenny-intel** [comment]
> @lenny-intel do we have any decision about the container naming ?

No, we didn't discuss in Core WG. Will be sure to discuss this week.

**Pull request merged**